pkg/controllers/user/pipeline/controller/pipeline/pipeline.go

```diff
@@ -47,27 +49,49 @@ func Register(ctx context.Context, cluster *config.UserContext) {
 
 func (l *Lifecycle) Create(obj *v3.Pipeline) (*v3.Pipeline, error) {
 
+	return l.sync(obj)
+}
+
+func (l *Lifecycle) Updated(obj *v3.Pipeline) (*v3.Pipeline, error) {
+
+	return l.sync(obj)
+}
+
+func (l *Lifecycle) Remove(obj *v3.Pipeline) (*v3.Pipeline, error) {
+
+	if obj.Status.WebHookID != "" {
+		if err := l.deleteHook(obj); err != nil {
+			//merely log error to avoid deletion block
+			logrus.Warningf("fail to delete previous set webhook for pipeline '%s' - %v", obj.Spec.DisplayName, err)
```

Comment:
Nit: use Warnf. Also is situation when previous webhook deletion failed, recoverable? What are the implication of the failed deletion?

Reply:
I may not be recoverable. For example when the user already deleted the webhook in this repo manually. So if the deletion failed we delete we proceed anyway.